functional/pipeline.py

```diff
@@ -1724,3 +1732,64 @@ def _wrap(value):
         return Sequence(value)
     else:
         return value
+
+def extend(func=None, aslist=False, final=False):
+    """
+    Function decorator for adding new methods to the Sequence class.
+
+    >>> @extend
```

Comment:
Its been a while since I've worked with decorators, but does it matter this is `@extend` versus `@extend()`? Does it make sense that for consistency to use the version with `@extend()` in all cases/examples?

Reply:
I think `@extend` is cleaner as some may forget to include brackets. It is also suggested to do so in this book (http://shop.oreilly.com/product/0636920027072.do).